src/main/java/ir/sahab/dockercomposer/DockerCompose.java

```diff
@@ -108,39 +111,51 @@ public Service getServiceByName(String name) {
     }
 
     /**
-     * A specification for running a set of services in a single docker-compose file using
-     * docker-compose utility.
+     * A specification for running a set of services in a single docker-compose file using docker-compose utility.
      */
     private static class DockerComposeSpec {
+
         Path file;
         String projectName;
         boolean forceRecreate = false;
         boolean forceDown = false;
         List<StartupCallback> startupCallbacks = new ArrayList<>();
         DockerComposeRunner runner; //  Runner of this spec
         Map<String, String> environment = new HashMap<>();
+
+        @Override
+        public String toString() {
+            return new ToStringBuilder(this)
+                    .append("file", file)
+                    .append("projectName", projectName)
+                    .append("forceRecreate", forceRecreate)
+                    .append("forceDown", forceDown)
+                    .append("startupCallbacks", startupCallbacks)
+                    .append("runner", runner)
+                    .append("environment", environment)
+                    .toString();
```

Comment:
Some fields like startupCallbacks, runner and projectName do not provide important information but makes the logs verbose. I think it is better not to include them.

Reply:
I think startupCallbacks have good information like WaitFor#PortOpenChecker which is one of the implementations of StartupCallback.